File: output/app/lib/Product.model.dart
class Product {
  
  String name;
  
  double price;
  
  String description;
  

  Product({
    
    required this.name,
    
    required this.price,
    
    required this.description,
    
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      
      name: json['name'],
      
      price: json['price'],
      
      description: json['description'],
      
    );
  }

  Map<String, dynamic> toJson() {
    return {
      
      'name': name,
      
      'price': price,
      
      'description': description,
      
    };
  }
}

File: src/models/product.model.ts
import {Entity, model, property} from '@loopback/repository';

@model({settings: {strict: false, mongodb: {collection: 'Products'}}})
export class Product extends Entity {
  @property({
    type: 'string',
    id: true,
    generated: true,
  })
  id?: string;

  @property({
    type: 'string',
    required: true,
  })
  name: string;

  @property({
    type: 'number',
    required: true,
  })
  price: number;

  @property({
    type: 'string',
  })
  description?: string;

  constructor(data?: Partial<Product>) {
    super(data);
  }
}
export type ProductWithRelations = Product;
